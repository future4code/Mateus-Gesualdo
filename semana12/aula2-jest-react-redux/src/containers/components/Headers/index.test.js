import React from 'react'
import renderer from 'react-test-renderer'
import { Headers, Button } from './index'
import { shallow } from 'enzyme'

describe('Testa o componente Headers', () => {
    test('snapshot', () => {
        const tree = renderer.create(<Headers/>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Chama a funcao goBack ao clicar no botao', () => {
		
		const goBack = jest.fn()
		const component = shallow(<Button onClick={goBack}/>)

		component.find(Button).simulate('click')

		expect(goBack).toHaveBeenCalledTimes(1)
	})
})