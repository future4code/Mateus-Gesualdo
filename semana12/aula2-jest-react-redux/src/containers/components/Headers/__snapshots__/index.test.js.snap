// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa o componente Headers snapshot 1`] = `
<div
  className="sc-AykKC sSVKO"
>
  <div>
    <img
      alt=""
      src="f4-icon.png"
    />
    <span
      className="sc-AykKD bipBvk"
    >
      4eddit
    </span>
  </div>
  <div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Voltar
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Sair
      </span>
    </button>
  </div>
</div>
`;
